packages/sources/intrinio/src/config.ts

```diff
@@ -1,36 +1,65 @@
-import { util } from '@chainlink/ea-bootstrap'
+import { Requester, Validator } from '@chainlink/ea-bootstrap'
+import { MakeWSHandler, Config, AdapterRequest } from '@chainlink/types'
+import { customParams } from './adapter'
+import IntrinioRealtime from 'intrinio-realtime'
 
 /**
  * @swagger
  * securityDefinitions:
  *  environment-variables:
- *    API_PROVIDER:
- *      required: false
  *    API_KEY:
  *      required: true
- *    SYMBOLS:
- *      required: true
  */
 
 export const NAME = 'INTRINIO'
 
-export const PROVIDER_OPTIONS = ['iex', 'quodd', 'fxcm']
+export const DEFAULT_API_ENDPOINT = 'https://api-v2.intrinio.com/'
+// const DEFAULT_WS_API_ENDPOINT = ''
 
-export type Config = {
-  key: string
-  symbols: string
-  provider: string
+export const makeConfig = (prefix?: string): Config => {
+  const config = Requester.getDefaultConfig(prefix, true)
+  config.api.baseURL = config.api.baseURL || DEFAULT_API_ENDPOINT
+  return config
 }
 
-export const makeConfig = (prefix?: string): Config => {
-  let provider = util.getEnv('API_PROVIDER', prefix) || ''
-  if (!PROVIDER_OPTIONS.includes(provider)) {
-    provider = PROVIDER_OPTIONS[0]
+export const makeWSHandler = (config?: Config): MakeWSHandler => {
+  // https://github.com/intrinio/intrinio-realtime-node-sdk
+
+  const getBase = (input: AdapterRequest): string => {
+    const validator = new Validator(input, customParams)
+    if (validator.error) {
+      return ''
+    }
+    return validator.overrideSymbol(NAME).toUpperCase()
   }
 
-  return {
-    key: util.getRequiredEnv('API_KEY', prefix),
-    symbols: util.getRequiredEnv('SYMBOLS', prefix),
-    provider: provider.toLowerCase(),
+  return () => {
+    const defaultConfig = config || makeConfig()
+
+    const ws = new IntrinioRealtime({
+      api_key: defaultConfig.apiKey,
+      provider: 'iex',
+    })
+
```

Comment:
Instead of the init method, we could ask for the token here. As this will be called on every request, we'll probably want to return a previous token if it's not expired. If it expired, we fetch it again

Reply:
We should make `makeWSHandler` async, fetch the token, and return the handler.